### PR TITLE
Remove unneeded plugins sections from the IO (inherited from parent)

### PR DESCRIPTION
**Pull request by iemejia**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by iemejia** [comment]
R: @jbonofre

**Comment by iemejia** [comment]
I just mention @dhalperi too since this comes from our discussion during the review of HBaseIO

**Comment by jbonofre** [comment]
Thanks, I'm reviewing.

File: sdks/java/io/google-cloud-platform/pom.xml [hunk 1]
```diff
@@ -39,21 +39,13 @@
     <plugins>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-compiler-plugin</artifactId>
-      </plugin>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
```
Comment (jbonofre) [inline sdks/java/io/google-cloud-platform/pom.xml]: just for the clarity, I would keep the `<groupId/>`.

**Comment by iemejia** [comment]
So what do you prefer?, should I put the three plugins in io/pom.xml ? (I had it like this, but I removed those when I tested and saw that worked as well without them).

File: sdks/java/io/google-cloud-platform/pom.xml (hunk 1 above)
Comment (iemejia) [inline sdks/java/io/google-cloud-platform/pom.xml]: The groupId is not removed, it is still there, it uses the one of the previous line.

**Comment by jbonofre** [comment]
No, what you did is correct and good !

File: sdks/java/io/google-cloud-platform/pom.xml (hunk 1 above)
Comment (jbonofre) [inline sdks/java/io/google-cloud-platform/pom.xml]: My bad, I missed this. Thanks.

**Comment by iemejia** [comment]
Perfect, thanks JB!

**Comment by iemejia** [comment]
It seems to be ok to merge.

**Comment by jbonofre** [comment]
Yes, gonna merge.

**Pull request closed without merging**